app/models/users/scopes/find_by_login.rb

```diff
@@ -0,0 +1,42 @@
+#-- encoding: UTF-8
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2020 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2017 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See docs/COPYRIGHT.rdoc for more details.
+#++
+
+# Find a user account by matching the exact login and then a case-insensitive
+# version. Exact matches will be given priority.
+module Users::Scopes
+  class FindByLogin
+    def self.fetch(login)
+      # First look for an exact match
+      user = User.where(login: login)
```

Comment:
This should probably be `User.find_by(..)` right?

Also, why not combine this in one statement? The first condition should take precendence, right? Also, there should never be the same login attribute twice since we do `validates_uniqueness_of :login, case_sensitive: false`

```
User.where(login: login').or(User.where("lower(login) = ?", login.to_s.downcase))
```

Reply:
Right you are of course. I simplified the code to only the second case.